src/language-markdown/printer-markdown.js

```diff
@@ -526,21 +567,247 @@ function getAncestorNode(path, typeOrTypes) {
+ * @typedef {import("./utils.js").WhitespaceValue} WhitespaceValue
+ * @typedef {{next?: TextNode | undefined | null, previous?: TextNode | undefined | null}} AdjacentNodes
+ * @typedef {import("./utils.js").WordKind} WordKind
+ * @typedef {import("../common/ast-path.js").default} AstPath
+ */
+
+/**
+ * Checks if given node can be converted to Space
+ *
+ * For example, if you would like to squash the multi-line string `"You might want\nto use Prettier."` into a single line,
+ * you would replace `"\n"` with `" "`. (`"You might want to use Prettier."`)
+ *
+ * However, you should note that Chinese and Japanese does not use U+0020 Space to divide words, so U+000A End of Line must not be replaced with it.
+ * Behavior in other languages (e.g. Thai) will not be changed because there are too much things to consider. (PR welcome)
+ *
+ * @param {AstPath} path path of given node
+ * @param {WhitespaceValue} value value of given node (typically `" "` or `"\n"`)
+ * @param {AdjacentNodes | undefined} adjacentNodes adjacent sibling nodes of given node
+ * @returns {boolean} `true` if given node can be converted to space, `false` if not (i.e. newline or empty character)
+ */
+function canBeConvertedToSpace(path, value, adjacentNodes) {
+  // "\n" or " ", of course " " always can be converted to Space
+  if (value !== "\n") {
+    return true;
+  }
+  // no adjacent nodes
+  if (!adjacentNodes) {
+    return true;
+  }
+  // e.g. " \nletter"
+  if (!adjacentNodes.previous || !adjacentNodes.next) {
+    return true;
+  }
+  const previousKind = adjacentNodes.previous.kind;
+  const nextKind = adjacentNodes.next.kind;
+  // "\n" between not western or Korean (han, kana, CJK punctuations) characters always can converted to Space
+  // Korean hangul simulates latin words; see #6516 (https://github.com/prettier/prettier/issues/6516)
+  if (
+    isWesternOrKoreanLetter(previousKind) &&
+    isWesternOrKoreanLetter(nextKind)
+  ) {
+    return true;
+  }
+  // han & hangul: same way preferred
+  if (
+    (previousKind === KIND_K_LETTER && nextKind === KIND_CJ_LETTER) ||
+    (nextKind === KIND_K_LETTER && previousKind === KIND_CJ_LETTER)
+  ) {
+    return true;
+  }
+  // Do not convert it to Space when:
+  if (
+    // Shall not be converted to Space around CJK punctuation
+    previousKind === KIND_CJK_PUNCTUATION ||
+    nextKind === KIND_CJK_PUNCTUATION ||
+    // "\n" between CJ always SHALL NOT be converted to space
+    // "\n" between Korean and CJ is better not to be converted to space
+    (isCJK(previousKind) && isCJK(nextKind))
+  ) {
+    return false;
+  }
+  const previousLastChar = adjacentNodes.previous.value?.at(-1);
+  const nextFirstChar = adjacentNodes.next.value?.at(0);
+  // The following rules do not precede the above rules (`return false`).
+  //
+  // Cases:
+  // - CJ & non-CJK
+  //
+  // 1. "\n" between special signs and CJ characters
+  // [corresponding sign][Space][any string][CJ][[\n]][target sign]
```

Comment:
This comment isn't clear. What is corresponding sign and why does it matter for this condition?

Reply:
I just forgot to update it.